crates/gosub_cairo/src/scene.rs

```diff
@@ -0,0 +1,232 @@
+        if let SceneCommand::Group { ref mut children, .. } = &mut self.root {
+            children.push(SceneCommand::simple_text(text.to_string(), pos, size));
+        }
+    }
+
+    fn apply_scene(&mut self, scene: &<CairoBackend as RenderBackend>::Scene, transform: Option<GsTransform>) {
+        if let SceneCommand::Group { ref mut children, .. } = &mut self.root {
+            children.push(SceneCommand::Group {
+                children: vec![scene.root.clone()],
+                transform: transform.unwrap_or(GsTransform::IDENTITY),
+            });
+        }
+    }
+
+    fn reset(&mut self) {
+        self.root = SceneCommand::new_group();
+    }
+
+    fn new() -> Self {
+        Self {
+            root: SceneCommand::new_group(),
+        }
+    }
+}
+
+/// Draws a rounded rectangle with specified border radii.
+pub fn draw_rounded_rect(cr: &cairo::Context, x: FP, y: FP, width: FP, height: FP, radius: &GsBorderRadius) {
+    // Helper function to get radius dimensions
+    let extract_radius = |r: &Radius| match r {
+        Radius::Uniform(r) => (*r, *r),
+        Radius::Elliptical(rx, ry) => (*rx, *ry),
+    };
+
+    let (tl_rx, tl_ry) = extract_radius(&radius.top_left);
+    let (tr_rx, tr_ry) = extract_radius(&radius.top_right);
+    let (bl_rx, bl_ry) = extract_radius(&radius.bottom_left);
+    let (br_rx, br_ry) = extract_radius(&radius.bottom_right);
+
+    // Start in the top-left corner, adjusted for the top-left radius
+    cr.move_to((x + tl_rx) as f64, y as f64);
+
+    // Top edge and top-right corner
+    cr.line_to((x + width - tr_rx) as f64, y as f64);
+    if tr_rx > 0.0 && tr_ry > 0.0 {
+        cr.arc(
+            (x + width - tr_rx) as f64,
+            (y + tr_ry) as f64,
+            tr_rx.min(tr_ry) as f64,
+            -std::f64::consts::FRAC_PI_2,
+            0.0,
+        );
+    }
+
+    // Right edge and bottom-right corner
+    cr.line_to((x + width) as f64, (y + height - br_ry) as f64);
+    if br_rx > 0.0 && br_ry > 0.0 {
+        cr.arc(
+            (x + width - br_rx) as f64,
+            (y + height - br_ry) as f64,
+            br_rx.min(br_ry) as f64,
+            0.0,
+            std::f64::consts::FRAC_PI_2,
+        );
+    }
+
+    // Bottom edge and bottom-left corner
+    cr.line_to((x + bl_rx) as f64, (y + height) as f64);
+    if bl_rx > 0.0 && bl_ry > 0.0 {
+        cr.arc(
+            (x + bl_rx) as f64,
+            (y + height - bl_ry) as f64,
+            bl_rx.min(bl_ry) as f64,
+            std::f64::consts::FRAC_PI_2,
+            std::f64::consts::PI,
+        );
+    }
+
+    // Left edge and top-left corner
+    cr.line_to(x as f64, (y + tl_ry) as f64);
+    if tl_rx > 0.0 && tl_ry > 0.0 {
+        cr.arc(
+            (x + tl_rx) as f64,
+            (y + tl_ry) as f64,
+            tl_rx.min(tl_ry) as f64,
+            std::f64::consts::PI,
+            3.0 * std::f64::consts::FRAC_PI_2,
+        );
+    }
+
+    cr.close_path();
+}
```

Comment:
It seems like this isn't used? There is another impl of that in `elements/border.rs` I assume this is for later, just pointing it out in case one of them should be removed.